src/main.cc

```diff
@@ -12,14 +12,50 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <fstream>
 #include <iostream>
+#include <string>
 
 #include "class.h"
 
-int main(int argc, char *argv[]) {
-  // TODO: read images given as arguments
-  // TODO: create an animation
-  // TODO: write the animation to disk
-  std::cout << "Compressing: hll wrld" << std::endl;
+// Return true on success or false on failure.
+static bool ReadImage(const char filename[], WebPPicture* const pic) {
+  const uint8_t* data = NULL;
+  size_t data_size = 0;
+  WebPImageReader reader;
+  bool ok;
+
+  if (!ImgIoUtilReadFile(filename, &data, &data_size)) return false;
+  reader = WebPGuessImageReader(data, data_size);
+  ok = reader(data, data_size, pic, 1, NULL);
+
+  free((void*)data);
+  return ok;
+}
+
+int main(int argc, char* argv[]) {
+  libwebp::Thumbnailer thumbnailer = libwebp::Thumbnailer();
+
+  WebPData webp_data;
+  WebPDataInit(&webp_data);
+
+  std::ifstream input_list(argv[1]);
+  const char* output = argv[2];
+  std::string filename_str;
+  int timestamp_ms;
```

Comment:
please declare those in the for loop (usually declare variables in the smallest scope possible).

Reply:
Does that mean I should use a `while (true)` loop for this case?